### PR TITLE
feat: send cache request to peers after PUT request

### PR DESCRIPTION
**Pull request by matthias-wright**
Each node is responsible for replicating its cached content onto its connected peers. After receiving a PUT request, a node will send a caching request to its connected peers.

### CODE REVIEW

File: crates/ursa-network/src/tests/service_tests.rs [hunk 1]
```diff
@@ -415,4 +416,63 @@ mod tests {
...
+            {
+                info!("[SwarmEvent::ConnectionEstablished]: {peer_id:?}, {peer_id_1:?}: ");
+                break;
+            }
+        }
+
+        let node_1_sender = node_1.command_sender();
+        tokio::task::spawn(async move { node_1.start().await.unwrap() });
+
+        let (sender, receiver) = oneshot::channel();
```
Comment (kckeiks) [inline crates/ursa-network/src/tests/service_tests.rs:443]:
```suggestion
        let (sender, _) = oneshot::channel();
```

File: crates/ursa-network/src/service.rs [hunk 2]
```diff
@@ -676,7 +677,15 @@ where
 
                 println!("cosmos");
             }
-            NetworkCommand::Put { cid: _, sender: _ } => (),
+            NetworkCommand::Put { cid, sender } => {
+                let swarm = self.swarm.behaviour_mut();
+                for peer in &self.peers {
+                    swarm
+                        .request_response
+                        .send_request(peer, UrsaExchangeRequest(RequestType::CacheRequest(cid)));
```
Comment (kckeiks) [inline crates/ursa-network/src/service.rs:685]: Are we handling this request or is that coming in a different PR?

Comment (matthias-wright) [reply crates/ursa-network/src/service.rs:685]: Yeah, that is coming in a different PR.

File: crates/ursa-rpc-service/src/api.rs [hunk 3]
```diff
@@ -216,6 +216,18 @@ where
...
+        let (sender, receiver) = oneshot::channel();
+        let request = NetworkCommand::Put {
+            cid: root_cid,
+            sender,
+        };
+        self.network_send.send(request)?;
+        if let Err(e) = receiver.await {
+            return Err(anyhow!(format!(
+                "The PUT failed, please check server logs {e:?}"
+            )));
```
Comment (scott-dn) [inline crates/ursa-rpc-service/src/api.rs:228]: just a minor one, for shorten.
```suggestion
            bail!("The PUT failed, please check server logs {e:?}");

```

**Comment by matthias-wright** [comment]
Closed because of #193

**Pull request closed without merging**